File: code/java/dayOfTheWeek.java
class Solution {
    public String dayOfTheWeek(int day, int month, int year) {
        
        //1971-01-01 Friday
        
        String[] day_of_week = {"Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"};
        
        int days = day - 1;
        int[] months = {31,28,31,30,31,30,31,31,30,31,30,31};
        days += 365 * (year-1971);
        days = days % 7 ;
        for(int i=0;i<month-1;i++)
            days += months[i];

        if(year >= 1972){
            days += (year - 1972) / 4 + 1;
            if(year % 4 == 0 && month < 3)
                days -- ;
            }     

        days = ((days % 7) + 6) % 7;
        if(days == 0)
            days = 7;
        return day_of_week[days-1];
    }
}

File: code/python/dayOfTheWeek.py
class Solution:
    def dayOfTheWeek(self, day: int, month: int, year: int) -> str:
        #1971-01-01 Friday
        
        day_of_week = ["Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"]
        
        days = day - 1
        months = [31,28,31,30,31,30,31,31,30,31,30,31]
        days += 365 * (year-1971)
        days = days % 7 
        for i in range(month-1):
            days += months[i]

        if year >= 1972:
            days += (year - 1972)//4 + 1
            if year % 4 == 0 and month < 3:
                days -= 1


        days = ((days % 7) + 6) % 7
        return day_of_week[days-1]
        
        
        
